rest/api_test.go

```diff
@@ -1686,20 +1685,230 @@ func TestReadChangesOptionsFromJSON(t *testing.T) {
 	assert.Equals(t, options.HeartbeatMs, uint64(60000))
 }
 
-func TestAccessControl(t *testing.T) {
-	restTester := initRestTester(db.IntSequenceType, `function(doc) {channel(doc.channels);}`)
-	defer restTester.Close()
-	testAccessControl(t, restTester)
-}
+func TestAllDocsAccessControl(t *testing.T) {
```

Comment:
While we're here, can you add a high level description of what behavior the test is validating?

Reply:
Would this test fail before the fix?

Reply:
This test didn't fail before the fix, but validates the change made in this PR to make sequence the primary id in the AllDocs index.  

The changes to TestChangesViewBackfillStarChannel cover the original bug - with those changes, the test failed prior to the fix.